### DISCUSSION TITLE
Welcome to Calibration-Shapes Discussions!

### DISCUSSION THREAD

**Discussion by 5axes** [5axes/Calibration-Shapes#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by BurtHarris** [comment]
My name is Burt Harris.  

I found project this because I'm making some changes to my 3-year-old Monoprice Select Mini, e.g., replacing the build surface, using a genuine E3D heater block, and installing an all-metal hot end.   I want to print with tolerances on the order of 0.1-0.2 mm (including holes.)  Knowing how good my tolerances are will help to design parts with appropriate clearance.

I bought a new set of calipers, equipped with Bluetooth, and experimented with recording measurements into Excel on my phone.   One idea I've got is to create something (perhaps an Excel spreadsheet) that takes care of processing a set of measurements and calculating new calibration settings, both for the printer (M92) and Cura (Horizontal Expansion and the new Hole Expansion).

I recently decided to call myself a "maker" rather than "retired."  I've got decades as a software developer but never wrote any python.   Perhaps this could be an opportunity to learn...

**Reply by 5axes** [reply]
Hello,
This type of solution already exist , have a look to the calibration cross : [https://www.thingiverse.com/thing:2484766](https://www.thingiverse.com/thing:2484766)
You will find associate with the geometry an Excel sheet. I'm using this solution. 

I haven't added this geometry to my plugin because : 
- You need to use an excel file to calculate the step values , 
-  It's depend of your machine settings
-  Usualy it's a test you make just one time , so  you don't need to have access to this geometry very oftenly

But it's a very good solution

**Reply by BurtHarris** [reply]
Thanks.  Yes, that's the idea.   The biggest thing that Calibration Cross is missing seems to be the calibration of hole sizes (inside measurements.)   Until V4.6, Cura apparently didn't have a setting for that, but now it has "horizontal expansion," "hole horizontal expansion," and "initial layer horizontal expansion".   This new parameter should be a big improvement.   

The cross is also exceeded my print volume.   I have some ideas on an alternate calibration geometry.   I'll try to firm those ideas up and run them by you if you are interested.   I agree the steps/mm part is likely to remain stable, but the expansion parameters seem like they might be more sensitive to other variables.

**Reply by 5axes** [reply]
For hole test I'm using a design very close of this part : [https://www.thingiverse.com/thing:2380801](https://www.thingiverse.com/thing:2380801)

https://github.com/5axes/Calibration-Shapes/wiki/Hole-Test

**Reply by BurtHarris** [reply]
That's pretty.   For calibration, it seems to me like measurement error is the enemy, and averaging measurements that should be identical is the best defense.  Also, inside measurements across circles are hard to make repeatable.   Have a look at this design: https://cad.onshape.com/documents/254b947c9b0e80f791f9fe0e/w/d5ad451b3a34535380b19906/e/773429aed4f97fa40b460cdb

**Reply by 5axes** [reply]
I m sorry but I don't have a onshape account to get the access to this part

**Comment by G4GNX** [comment]
I'm not sure if this is the right place for a possible bug report, so please excuse if not.
I'm having issues with the Temperature Tower, where the script appears to be generating the wrong Gcode. I've also seen a similar issue reported by someone on Facebook.

The issue:

I set the start temp to 220 and interval of 5 degrees.
It started off fine at 220 and dropped the temperature correctly as the tower rose, but when it got to 200 it stopped extruding but all other movements were carried out. I happened to be watching this on the Octoprint terminal and the printer was constantly sending error messages that it had prevented extrusion at too low temperature which was configured to 175.
I stopped the print and checked the Gcode and found that the parts beyond 200 were missing except for a final section at 175 degrees.
I tried re-slicing and got a different result, with the missing sections restored, but with a final section at 175 degrees. There should only be 9 sections printed starting at 220 dropping 5 degrees each time and ending at 180, the rogue 175 degree section is number 10 and should not even be there.
I gave the Python code (used to generate the script) a cursory glance late last night and I can't see why it would do that.
My latest Gcode compile is attached below:
[Cura Temperature Tower 5.zip](https://github.com/5axes/Calibration-Shapes/files/6212487/Cura.Temperature.Tower.5.zip)

**Reply by 5axes** [reply]
Your G-Code looks correct. I Think it's not an issue on the script .  I think it's much more an issue on your settings.
![image](https://user-images.githubusercontent.com/11015345/112662180-af57d500-8e57-11eb-8fc9-65fb4bb54f1b.png)

**Reply by G4GNX** [reply]
I'm quite happy to admit if I got a setting wrong, but I can't understand why it appears to be trying to print 10 segments, or does that depend on the over-all calculated height?
My settings are below and my layer height is 0.2mm
![Cura](https://user-images.githubusercontent.com/60974988/112676284-27c29400-8e60-11eb-88e9-1362957485a8.jpg)

**Reply by 5axes** [reply]
The initial design was made for a 0.16 Layer Height so this is the origine of your error.  we don't get a constant height with 38 as increment. I should think aboiut the possibility to define a diffrent tower in case of layer height of 0.2 ... next release
